proteuslib/unit_models/tests/test_reverse_osmosis_0D.py

```diff
@@ -175,6 +175,74 @@ def test_option_pressure_change_calculated():
     assert isinstance(m.fs.unit.spacer_porosity, Var)
     assert isinstance(m.fs.unit.N_Re_io, Var)
 
+class Test_initialization_args():
+    @pytest.fixture(scope="class")
+    def m(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+
+        m.fs.properties = props.NaClParameterBlock()
+
+        m.fs.unit = ReverseOsmosis0D(default={
+            "property_package": m.fs.properties,
+            "has_pressure_change": True,
+            "concentration_polarization_type": ConcentrationPolarizationType.calculated,
+            "mass_transfer_coefficient": MassTransferCoefficient.calculated,
+            "pressure_change_type": PressureChangeType.calculated
+        })
+
+        # fully specify system
+        feed_flow_mass = 1000 / 3600
+        feed_mass_frac_NaCl = 0.034283
+        feed_pressure = 70e5
+
+        feed_temperature = 273.15 + 25
+        A = 4.2e-12
+        B = 3.5e-8
+        pressure_atmospheric = 1e5
+        feed_mass_frac_H2O = 1 - feed_mass_frac_NaCl
+
+        m.fs.unit.inlet.flow_mass_phase_comp[0, 'Liq', 'NaCl'].fix(
+            feed_flow_mass * feed_mass_frac_NaCl)
+
+        m.fs.unit.inlet.flow_mass_phase_comp[0, 'Liq', 'H2O'].fix(
+            feed_flow_mass * feed_mass_frac_H2O)
+
+        m.fs.unit.inlet.pressure[0].fix(feed_pressure)
+        m.fs.unit.inlet.temperature[0].fix(feed_temperature)
+        m.fs.unit.A_comp.fix(A)
+        m.fs.unit.B_comp.fix(B)
+        m.fs.unit.permeate.pressure[0].fix(pressure_atmospheric)
+        m.fs.unit.N_Re_io[0, 'in'].fix(400)
+        m.fs.unit.recovery_mass_phase_comp[0, 'Liq', 'H2O'].fix(0.5)
+        m.fs.unit.spacer_porosity.fix(0.97)
+        m.fs.unit.channel_height.fix(0.001)
+
+        return m
+
+    @pytest.mark.component
+    def test_initialize_ignore_dof(self, m):
+         # Unfix a variable and check that ignore_dof arg works
+        temp = m.fs.unit.N_Re_io[0, 'in'].value
+        m.fs.unit.N_Re_io[0, 'in'].unfix()
+        # Next line should pass since we are ignoring non-zero DOF
+        m.fs.unit.initialize(ignore_dof=True, fail_on_warning=False)
+        # Check that ValueError is thrown when ignore_dof and fail_on_warning are set to do so
+        with pytest.raises(ValueError):
```

Comment:
It is good practice to check the error message here to.

```
with pytest.raises(ValueError, match="expected message"):
```

Reply:
Thanks- I was trying to recall how to do this from memory but skipped over it.